### DISCUSSION TITLE
whobpyt dev discussion 2024-03-13

### DISCUSSION THREAD

**Discussion by JohnGriffiths** [GriffithsLab/whobpyt#183] [Meeting Minutes]
code base discussion

**Comment by JohnGriffiths** [comment]
Note re this line

https://github.com/GriffithsLab/whobpyt/blob/ef921c3d2e365149a9eff1e00d6a2817292d6503/whobpyt/datatypes/parameter.py#L16

if asRand = True
and prior_mean and prior_std are given
and fit_par is True
then it will sample from the distribution

**Reply by JohnGriffiths** [reply]
`lb` argument is lower bound

note :in general all params should be in the docstring

**Comment by JohnWangDataAnalyst** [comment]
https://github.com/GriffithsLab/whobpyt/blob/e37c249ec2343713152d390ef5240987be58a52a/whobpyt/datatypes/parameter.py#L52
test

**Comment by sbastiaens** [comment]
https://github.com/GriffithsLab/whobpyt/blob/e37c249ec2343713152d390ef5240987be58a52a/whobpyt/datatypes/parameter.py#L118

For the parameter lb (lower bound), default is equal to 0 now, but shouldn't it be 'None' as some parameters will be negative?

**Comment by JohnGriffiths** [comment]
if fit_par is specified then it will fit the hyperparameter, i.e. the mean and variance

**Comment by JohnGriffiths** [comment]
inputs go in as numpy variables, params converts them at the end to tensors

**Comment by P-Oveisi** [comment]
https://github.com/GriffithsLab/whobpyt/blob/e37c249ec2343713152d390ef5240987be58a52a/whobpyt/datatypes/parameter.py#L67

I think there should be an else clause stating that if asRand is false, then set the random seed as a fixed number. (still sampling from the distribution, but now in every run, the same sample would be chosen)
